rust/arrow/src/array/equal/utils.rs

```diff
@@ -76,3 +80,185 @@ pub(super) fn equal_len(
 ) -> bool {
     lhs_values[lhs_start..(lhs_start + len)] == rhs_values[rhs_start..(rhs_start + len)]
 }
+
+/// Computes the logical validity bitmap of the array data using the
+/// parent's array data. The parent should be a list or struct, else
+/// the logical bitmap of the array is returned unaltered.
+///
+/// Parent data is passed along with the parent's logical bitmap, as
+/// nested arrays could have a logical bitmap different to the physical
+/// one on the `ArrayData`.
+pub(super) fn child_logical_null_buffer(
+    parent_data: &ArrayData,
+    logical_null_buffer: Option<Buffer>,
```

Comment:
@alamb @jorgecarleitao I wanted to make this `Option<&Buffer>` to avoid cloning, but because I create a `Bitmap` for `parent_bitmap` and `self_null_bitmap` , I have to end up cloning the `&Buffer`. So it's extra work to change the signature, and probably doesn't yield any benefit.